File: .local/bin/inbox.py
#!/usr/bin/python3
"""Inject email directly into maildir."""
import mailbox
import email
import argparse
import os
from email import encoders
from email.mime.base import MIMEBase
import email.utils
from email.mime.text import MIMEText


def main():
    """Program entry."""
    parser = argparse.ArgumentParser()
    parser.add_argument("--address", type=str, default="enckse@voidedtech.com")
    parser.add_argument("--maildir", type=str, default="/home/enck/.mutt/imap/fastmail/Filtered/Automated/")
    parser.add_argument("--subject", type=str)
    parser.add_argument("--input", type=str)
    parser.add_argument("--plaintext", action="store_true")
    args = parser.parse_args()
    if not args.address or not args.subject:
        print("address and subject required")
        return
    if not args.maildir or not os.path.exists(args.maildir):
        print("maildir does not exist")
        return
    readmode = "r"
    if not args.plaintext:
        readmode += "b"
    with open(args.input, readmode) as f:
        if args.plaintext:
            msg = MIMEText(f.read())
        else:
            msg = MIMEBase('application', 'zip')
            msg.set_payload(f.read())
            encoders.encode_base64(msg)
            msg.add_header('Content-Disposition',
                           'attachment',
                           filename=args.input)
            message = mailbox.email.message_from_file(f)
    msg.add_header("To", args.address)
    msg.add_header("From", args.address)
    msg.add_header("Subject", args.subject)
    msg.add_header("Date", email.utils.formatdate(localtime=True))
    maildir = mailbox.Maildir(args.maildir)
    maildir.add(msg)


if __name__ == "__main__":
    main()

File: .local/lib/inbox.pl
#!/usr/bin/perl
use strict;
use warnings;
use autodie;

use Getopt::Long;
use Email::MIME;
use utf8;
use MIME::Entity;
use Mail::Message::Convert::MimeEntity;
use Mail::Box::Maildir;
use Email::Date::Format qw(email_date);

my $addr    = 'enckse@voidedtech.com';
my $maildir = "/home/enck/store/personal/imap/fastmail/Filtered/Automated/";
my $subject = "";
my $input   = "";
GetOptions(
    "address=s" => \$addr,
    "maildir=s" => \$maildir,
    "subject=s" => \$subject,
    "input=s"   => \$input,
) or die "unable to parse commands";

$addr                           || die "address is required";
$subject                        || die "subject is required";
$maildir                        || die "maildir is required";
( -e $maildir and -d $maildir ) || die "invalid mailrdir";
-e $input                       || die "invalid input file";

my $type;
my $message_body;

my $mime = MIME::Entity->build(
    Type    => "multipart/mixed",
    From    => $addr,
    To      => $addr,
    Subject => $subject,
    Date    => email_date,
    Data    => $message_body
);

$mime->attach(
    Path     => $input,
    Type     => "application/zip",
    Encoding => "base64"
);

my $convert = Mail::Message::Convert::MimeEntity->new;
my Mail::Message $msg = $convert->from($mime);

open STDERR, '>', '/dev/null';

my $folder = new Mail::Box::Maildir(
    folder            => $maildir,
    remove_when_empty => 0,
    access            => 'rw',
);

$folder->addMessage($msg);
